commons/src/main/java/fr/free/nrw/commons/upload/GPSExtractor.java

```diff
@@ -33,21 +69,67 @@ public String getCoords() {
         }
 
         if (exif.getAttribute(ExifInterface.TAG_GPS_LATITUDE) == null) {
-            Log.d("Image", "Picture has no GPS info");
-            return null;
-        }
-        else {
+            imageCoordsExists = false;
+            Log.d(TAG, "Picture has no GPS info");
+
+            //Check what user's preference is for automatic location detection
+            boolean gpsPrefEnabled = gpsPreferenceEnabled();
+
+            if (gpsPrefEnabled) {
+                //If pref enabled, get current location
+                Location location = locationManager.getLastKnownLocation(provider);
+
+                if (location != null) {
+                    myLocationListener.onLocationChanged(location);
```

Comment:
It seems a little funky to call this but it makes sense according to the LocationManager docs. This might be more logical to only call once, immediately after requestLocationUpdates(). As it's written, this would be called for every invocation of getCoords() even when the LocationManager was regularly calling into onLocationChanged().